packages/dataviews/src/dataviews-layouts/table/column-header-menu.tsx

```diff
@@ -84,11 +84,12 @@ const _HeaderMenu = forwardRef( function HeaderMenu< Item >(
 	operators = sanitizeOperators( field );
 	// Filter can be added:
 	// 1. If the field is not already part of a view's filters.
-	// 2. If the field meets the type and operator requirements.
-	// 3. If it's not primary. If it is, it should be already visible.
+	// 2. If the field has elements or Edit property.
+	// 3. If the field meets the type and operator requirements.
+	// 4. If it's not primary. If it is, it should be already visible.
 	canAddFilter =
 		! view.filters?.some( ( _filter ) => fieldId === _filter.field ) &&
-		!! field.elements?.length &&
+		!! ( field.elements?.length || field.Edit ) &&
```

Comment:
There are existing filters in dotcom sites, agencies (multiple places) and jetpack cloud that use this package directly. We need to be opt-in.

We also need to engage design to confirm displaying the edit controls in the existing filters is ok. Otherwise, we need to add the "disable edit" feature here and update those places, first.

Reply:
> There are existing filters in dotcom sites, agencies (multiple places) and jetpack cloud that use this package directly. We need to be opt-in.

In that case, IMO we should disable the edit by default even if `elements` prop is provided to avoid any issues.  Consumer should explicitly opt-in to this edit feature.

Reply:
Can you explain more @oandregal what issues this creates?

Reply:
I believe @oandregal's concern is that enabling input filters by default, particularly when fields include Edit controls, might lead to problems for existing users of this package. 

For example, dotcom sites currently use only one filter enabled by the elements prop. Upgrading to the new version might show more filters. André suggests [disabling input by default](https://github.com/Automattic/wp-calypso/pull/103276#discussion_r2092601407) or adding a property to disable it in this PR.

Reply:
> We also need to engage design to confirm displaying the edit controls in the existing filters is ok (may require updating the Edit for existing types, etc.).


@Automattic/dotcom-design 👋 This PR adds user input filters by using `Edit` controls for rendering the filters UI. The goal is to provide a way to filter data by user input. By using `Edit` controls, we can provide a more flexible and powerful way for customers to filter data, such as Advanced DateTime filters. Does this look good to you?

Here is video demo of the input filters:

https://github.com/user-attachments/assets/8617340e-622f-435f-91b1-9ae0002a61ff

And we will use this feature to build the filters (figma: hdK9XLiOefugy2bxalKBN1-fi-2795_100658) for WooCommerce Analytics.